Mage.Sets/src/mage/cards/t/ThousandYearStorm.java

```diff
@@ -85,13 +110,16 @@ public ThousandYearStormAbility copy() {
 }
 
 class ThousandYearStormEffect extends OneShotEffect {
+    private int stormCount;
```

Comment:
you might probably want to use game.getState().setValue() instead of storing it here to avoid some nasty rollback bugs

Reply:
I don't think this causes a bug at the moment because it is only used during the short time between a trigger going on the stack and resolving. Rollbacks only allow jumping back to the start of a turn. There shouldn't be a Thousand-Year Storm trigger on the stack at the time the rollback is jumping to.

That being said it would probably be nicer to have that number stored in the game state. What would you normally use as key for such a value? Is the id of the effect a valid choice?

If this is a bug the implementation of Sentinel Tower is probably bugged as well since it's implementation is pretty similar to my implementation

Reply:
I think rollbacks are also used with spell canceling, so it might be possible to increment this by casting and canceling one spell over and over. (see #6618, rollback was actually not the problem here, but the pattern could be similar). Its at least worth checking. If the fix is actually needed then yeah, effect id + card name probably will do the trick.